### PR TITLE
Add new test for deepcopy

### PR DESCRIPTION
**Pull request by spzala**
Add new test convering servicecatalog types related to changes made under
https://github.com/kubernetes/gengo/pull/46

### CODE REVIEW

**Comment by thockin** [comment]
I appreciate the followup.  Can we narrow this down and rename it to reflect the type that is being demonstrated.  Specifically, a struct with a single member field that triggers the rendered code - that way each output test is trivial to look at and decide if changes are correct.  It might mean 3 or 4 individual new dirs..

**Comment by spzala** [comment]
@thockin - yrw and sure, that makes sense. I will update the changes. Thanks!!

**Comment by spzala** [comment]
@thockin - Hi Tim, as you could tell I am new to Go, but have tried to identify new types and added tests for them. Hope that makes sense. Thanks!!

File: examples/deepcopy-gen/output_tests/interfaces/doc.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+// This is a test package.
+package interfaces
+
+type Inter interface {
+  function() float64
+}
+
+type Ttest struct {
+  i []Inter
```
Comment (thockin) [inline examples/deepcopy-gen/output_tests/interfaces/doc.go]: I think this has to be named `I` - non-private name.

File: examples/deepcopy-gen/output_tests/interfaces/doc.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:deepcopy-gen=package
+
+// This is a test package.
+package interfaces
+
+type Inter interface {
```
Comment (thockin) [inline examples/deepcopy-gen/output_tests/interfaces/doc.go]: What does `Inter` mean?

File: examples/deepcopy-gen/output_tests/pointer/zz_generated.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+func DeepCopy_pointer_Ttest(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
+		in := in.(*Ttest)
+		out := out.(*Ttest)
+		*out = *in
+		if in.Types != nil {
+			in, out := &in.Types, &out.Types
+			*out = make(map[string]*Ttest)
+			for key, val := range *in {
+				newVal, err := c.DeepCopy(&val)
```
Comment (thockin) [inline examples/deepcopy-gen/output_tests/pointer/zz_generated.go]: Why do we expect this to hit the Cloner path?

**Comment by thockin** [comment]
LGTM!

**Pull request merged**